runtime/testing/jvm/src/aws/smithy/kotlin/runtime/testing/RandomInputStream.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+package aws.smithy.kotlin.runtime.testing
+
+import java.io.IOException
+import java.io.InputStream
+import kotlin.random.Random
+
+/**
+ * Test utility InputStream implementation that generates random ASCII data when
+ * read, up to the size specified when constructed.
+ */
+public class RandomInputStream constructor(
+    /** The requested amount of data contained in this random stream.  */
+    private val lengthInBytes: Long,
+
+    /** Flag controlling whether binary or character data is used.  */
+    private val binaryData: Boolean = false
+) : InputStream() {
```

Comment:
**Nit**: I don't think the KDoc directly on constructor parameters works (at least I couldn't get anything to show up in my IDE). I think those docs should be specified in `@param` tags on the class-level KDoc:

```kotlin
/**
 * Test utility InputStream implementation that generates random ASCII data when
 * read, up to the size specified when constructed.
 * @param lengthInBytes The requested amount of data contained in this random stream.
 * @param binaryData Flag controlling whether binary or character data is used.
 */
```